script/cluster-kind.mk

```diff
@@ -0,0 +1,24 @@
+# This file provides targets which create a local k8s cluster setup
+# with OIDC integration for development and testing.
+KUBE ?= ${HOME}/.kube
+CLUSTER_NAME ?= kubeapps
+
+CLUSTER_CONFIG = ${KUBE}/kind-config-${CLUSTER_NAME}
+
+devel/local-dev-apiserver-config.json:
+	cat docs/user/manifests/kubeapps-local-dev-apiserver-config.json | \
+	jq ".nodes[0].extraMounts[0].hostPath = \"${PWD}/script/test-certs/ca.cert.pem\"" > \
+	$@
+
+${CLUSTER_CONFIG}: devel/local-dev-apiserver-config.json
+	kind create cluster \
+		--name ${CLUSTER_NAME} \
+		--config=./devel/local-dev-apiserver-config.json \
+		--retain
+
+cluster-kind: ${CLUSTER_CONFIG}
```

Comment:
you also need a target to load the `kubeconfig-path`, no? `export KUBECONFIG=$(kind get kubeconfig-path --name=kind)`

Reply:
Makefiles don't let you change the external environment, each command is run in a subshell, which is a good thing... so what I want to do is separate the cluster creation helpers (like this one) from the kubeapps deployment configuration helper (which I'll add next). This is because deploying a configured oidc kubeapps example will be the same whether it's on a local kind or minishift or non-local env. So it'll be:

1) Create your configured cluster (ie. with the oidc endpoint)
2) Deploy the example kubeapps (which will configured for oidc)

Reply:
true, I forgot that

_I am not a enthusiast of Makefiles_

Reply:
> true, I forgot that
> 
> _I am not a enthusiast of Makefiles_

That's right - you mentioned and I wanted to hear more... next standup topic? :) (FWIW, I'm happy to use something else - I just don't see anything that fits the bill. Bash scripts need a lot of extra work to get the same benefits, imo, but I don't care that much).